tools/test2markdown/test2markdown.py

```diff
@@ -20,116 +20,190 @@
+        """Converts a record object to Markdown format.
+
+        Args:
+            record (Record_Object): Record object to convert
+            level (int): Current level of the record object
+
+        Returns:
+            Ret: Status
+        """
+
+        if record.n_typ.name == "Diagram":
+            self._print_diagram(record, level)
+
+        if record.n_typ.name == "Info":
+            self._print_info(record, level)
+
+        elif record.n_typ.name == "SwTestCase":
+            self._print_sw_test_case(record, level)
+
+        else:
+            # Skipped.
+            pass
+
+        return Ret.OK
+
+    def _print_table_head(self) -> None:
+        """Prints the table head for software requirements and constraints.
+        """
+        column_titles = ["Attribute", "Value"]
+        markdown_table_head = self.markdown_create_table_head(column_titles)
+
+        self._fd.write(markdown_table_head)
+
+    # pylint: disable=unused-argument
+    def _print_diagram(self, diagram: Record_Object, level: int) -> None:
+        """Prints the diagram.
+
+        Args:
+            diagram (Record_Object): Diagram to print
+            level (int): Current level of the record object
+        """
+        plantuml_generator = PlantUML()
+        image_format = "png"
+        diagram_dict = diagram.to_python_dict()
+        file_path = diagram_dict["file_path"]
+        caption = diagram_dict["caption"]
+        full_file_path = file_path
+        file_dst_path = None
+
+        # Is the path to the diagram invalid?
+        if os.path.isfile(full_file_path) is False:
+
+            full_file_path = self._locate_file(full_file_path)
+            # Diagram not found?
+            if full_file_path is None:
+                raise FileNotFoundError(f"{file_path} not found.")
+
+        if plantuml_generator.is_plantuml_file(file_path):
+
+            plantuml_generator.generate(
+                image_format, full_file_path, self._args.out)
+
+            file_dst_path = os.path.basename(full_file_path)
+            file_dst_path = os.path.splitext(file_dst_path)[0]
+            file_dst_path += "." + image_format
+
+            # PlantUML uses as output filename the diagram name if available.
+            # The diagram name may differ from the filename.
+            # To aovid that a invalid reference will be in the Markdown document,
+            # ensure that the generated filename is as expected.
+            expected_dst_path = os.path.join(self._args.out, file_dst_path)
+            if os.path.isfile(expected_dst_path) is False:
+                raise FileNotFoundError(
+                    f"{file_path} diagram name ('@startuml <name>') may differ from file name,"
+                    " expected {expected_dst_path}."
+                )
+
+        else:
+            # Copy diagram image file to output folder.
+            shutil.copy(full_file_path, self._args.out)
+            file_dst_path = os.path.basename(full_file_path)
```

Comment:
Add issue to refactor this to the base class. Every derivation clones this code.